src/components/SSOLanding.js

```diff
@@ -28,24 +29,35 @@ const SSOLanding = () => {
 
   const token = getToken();
 
+  const getTenant = () => {
+    const params = getParams();
+    const tenant = config.useSecureTokens
+      ? params?.tenant
+      : parseJWT(token)?.tenant;
+
+    return tenant || okapi.tenant;
+  };
+
+  const tenant = getTenant();
+
   useEffect(() => {
-    if (token) {
-      requestUserWithPermsDeb(okapi.url, store, okapi.tenant, token);
+    if (tenant) {
+      requestUserWithPermsDeb(okapi.url, store, tenant, token);
     }
-  }, [token, store]);
+  }, [tenant, token, store]);
 
-  if (!token) {
+  if (!tenant) {
     return (
       <div data-test-sso-error>
-        No <code>ssoToken</code> cookie or query parameter
+        No <code>tenant</code> in cookie or query parameter
       </div>
     );
   }
 
   return (
     <div data-test-sso-success>
       <p>
-        Logged in with token <tt>{token}</tt> from {getParams()?.ssoToken ? 'param' : 'cookie'}.
+        Logged in with tenant <tt>{tenant}</tt>.
```

Comment:
I would propose to show spinner here instead as for `useSecureTokens=true` we'll have no token available and message says nothing in general

Reply:
@NikitaSedyx's suggestion here is solid and would be a nice improvement over this unlocalized message (😬 ), but if you want to minimize changes to the existing implementation, I think your change is OK as-is.